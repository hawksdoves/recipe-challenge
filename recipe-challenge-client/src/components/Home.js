import React from 'react';

import getRecipes from '../services/getRecipes';
import RecipeSearch from './RecipeSearch';

class Home extends React.Component {

    constructor(props) {
        super(props)
        this.state = { recipes: [] };
    }

    componentDidMount() {
        getRecipes()
            .then(resp => resp.data ? this.setState({ recipes: resp.data.recipes }) : null)
    }

    render() {
        return (
            <div>
                <h1>Welcome to Recipes'R'Us </h1>
                { this.state.recipes.length > 0 ?
                    <RecipeSearch recipes={this.state.recipes} /> :
                    <div>Sorry, we currently have no recipes for you</div>
                }
            </div>
        )
    }
    
};

export default Home;