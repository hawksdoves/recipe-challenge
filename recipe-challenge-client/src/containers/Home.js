import React from 'react';
import { connect } from 'react-redux';
import Home from '../components/Home';
import getAllRecipes from '../actions/getRecipes';

const mapStateToProps = (state) => {
    return {
        recipes: state.recipes
    }
}

const HomeContainer = connect(
  mapStateToProps
)(Home)

export default HomeContainer