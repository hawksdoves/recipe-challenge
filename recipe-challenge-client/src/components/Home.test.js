import React from 'react';
import { shallow, mount } from 'enzyme';

import Home from './Home';

describe('Home', () => {
    
    it('should render all the recipes correctly', () => {
        const wrapper = shallow(<Home />);

        expect(wrapper).toMatchSnapshot()
    })

    it('should display error message if there are no recipes', () =>{

        const wrapper = shallow(<Home />);

        expect(wrapper).toMatchSnapshot()
    })
});
