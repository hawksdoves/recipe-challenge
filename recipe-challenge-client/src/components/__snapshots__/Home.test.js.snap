// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should display error message if there are no recipes 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Welcome to Recipes'R'Us 
    </h1>
    <div>
        Sorry, we currently have no recipes for you
    </div>
</div>,
  "nodes": Array [
    <div>
      <h1>
            Welcome to Recipes'R'Us 
      </h1>
      <div>
            Sorry, we currently have no recipes for you
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "recipes": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h1>
                    Welcome to Recipes'R'Us 
          </h1>
          <div>
                    Sorry, we currently have no recipes for you
          </div>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <h1>
                    Welcome to Recipes'R'Us 
          </h1>
          <div>
                    Sorry, we currently have no recipes for you
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home />,
}
`;

exports[`Home should render all the recipes correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Welcome to Recipes'R'Us 
    </h1>
    <div>
        Sorry, we currently have no recipes for you
    </div>
</div>,
  "nodes": Array [
    <div>
      <h1>
            Welcome to Recipes'R'Us 
      </h1>
      <div>
            Sorry, we currently have no recipes for you
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "recipes": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h1>
                    Welcome to Recipes'R'Us 
          </h1>
          <div>
                    Sorry, we currently have no recipes for you
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h1>
                    Welcome to Recipes'R'Us 
          </h1>
          <div>
                    Sorry, we currently have no recipes for you
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home />,
}
`;
